src/DynamoPackages/PackageLoader.cs

```diff
@@ -29,8 +30,15 @@ public class PackageLoader : LogSourceBase
     {
         internal event Action<Assembly> RequestLoadNodeLibrary;
         internal event Func<string, IEnumerable<CustomNodeInfo>> RequestLoadCustomNodeDirectory;
+        internal event Func<string, dynamic> RequestLoadExtension;
+        internal event Action<dynamic> RequestAddExtension;
         public event Action<Package> PackageAdded;
         public event Action<Package> PackageRemoved;
+        /// <summary>
+        /// A map of the extension manifest paths which the package loader found during package load
+        /// and has requested be loaded. The result of the load request is mapped to the requested path.
+        /// </summary>
+        public Dictionary<string, dynamic> RequestedExtensions { get; private set; } = new Dictionary<string, dynamic>();
```

Comment:
Can this be `internal`?

Reply:
if we expose `packageManagerExtension` internals to dynamocore

Reply:
actually I think we can just make it internal.